script.js

```diff
@@ -1,14 +1,349 @@
+   no accounts or sign-ups required.
+    has been the industry standard.`,
+    technologies: ['HTML', 'Boostrap', 'Ruby'],
+    button: 'See Project',
+  },
+  {
+    title: 'Website Portfolio',
+    image: 'images/Img Placeholder (5).png',
+    description: `A daily selection of privately 
+    personalized reads; no accounts or 
+    sign-ups required
+    personalized reads; no accounts or 
+    sign-ups required. Has been the
+     industry's standard.`,
+    technologies: ['HTML', 'Boostrap', 'Ruby'],
+    button: 'See Project',
+  },
+];
+
+const portfolio = [
+  {
+    title: 'My Recent Works',
+    header: 'Multi Post Stories',
+    technologies: ['CSS', 'HTML', 'Boostrap', 'Ruby'],
+    image: 'images/Img Placeholder.png',
+    imageMob: 'images/Img Placeholder (3).png',
+    description: `A daily selection
+    of privately personalized reads; 
+    no accounts or sign-ups <br> required. 
+    This has been the industry's 
+    standard dummy <br> text ever since 
+    the 1500s, when an unknown
+     printer took a <br>
+    standard dummy text.`,
+    button: 'See Project',
+  },
+];
+
+const popup = [
+  {
+    title: 'Keeping track of hundreds of components website',
+    titleMob: 'Multi Post Stories',
+    button: ['HTML', 'Boostrap', 'Ruby on rails'],
+    image: 'images/aya.png',
+    imageMob: 'images/snapsmall.png',
+    description: `Lorem Ipsum is simply dummy text of
+     the printing and typesetting industry. Lorem Ipsum has been the 
+     industry's standard dummy text ever since the 1500s, when an unknown
+     make a type specimen book.
+       It has survived not only five centuries,
+     but also the leap into electronic typesetting, remaining essent`,
+    descriptionMob: `Lorem Ipsum is simply dummy text of
+    the printing and typesetting industry. Lorem Ipsum has been the 
+    industry's standard dummy text ever since the 1500s, when an unknown
+      make a type specimen book.
+      It has survived not only five centuries,
+    but also the leap into electronic typesetting, remaining essent`,
+    links: ['See live', 'See source'],
+    linkImg: ['images/link.png', 'images/Vector.png'],
+    linkSource: ['https://blessman-newton.github.io/Portfolio/', 'https://github.com/Blessman-Newton/Portfolio'],
+
+  },
+];
+// JavaScript arrays containing work section information
+const projectsMobile = [
+  {
+    title: 'Profesional Art Printing Data',
+    image: 'images/Img Placeholder (2).png',
+    description: `A daily selection of privately personalized reads;
+    no accounts or sign-ups required.
+    has been the industry standard.`,
+    technologies: ['HTML', 'Boostrap', 'Ruby'],
+    button: 'See Project',
+  },
+  {
+    title: 'Profesional Art Printing Data',
+    image: 'images/Img Placeholder (2).png',
+    description: `A daily selection of privately personalized reads;
+    no accounts or sign-ups required.
+    has been the industry standard.`,
+    technologies: ['HTML', 'Boostrap', 'Ruby'],
```

Comment:
- [ ] To improve your code and implement the required interactions, we'll first refactor the projects section to use a JavaScript array to store all project information in objects. Then, we'll create a function to dynamically generate the projects section on the main page. Finally, we'll implement the popup window for both mobile and desktop screen sizes. Here's the updated code:

```html
<!-- HTML: Add an empty popup container in the body -->
<div class="popup-container" id="popupContainer"></div>
```

```javascript
// JavaScript: Create a function to generate the projects section dynamically
function generateProjectsSection(projectsData) {
  let desktopTemplate = '';
  let mobileTemplate = '';

  projectsData.forEach((project) => {
    let technologies = project.technologies.map((tech) => `<li>${tech}</li>`).join('');
    const desktopTag = `
      <div class="card">
        <img class="card-img" src="${project.image}" alt="">
        <div class="cont">
          <h2>${project.title}</h2>
          <p>${project.description}</p>
          <ul>${technologies}</ul>
        </div>
        <div class="card-button">
          <button type="button" class="view-project">${project.button}</button>
        </div>
      </div>
    `;
    desktopTemplate += desktopTag;

    const mobileTag = `
      <div class="card">
        <img class="card-img" src="${project.image}" alt="">
        <div class="cont">
          <h2>${project.title}</h2>
          <p>${project.description}</p>
          <ul>${technologies}</ul>
        </div>
        <div class="card-button">
          <button type="button" class="view-project">${project.button}</button>
        </div>
      </div>
    `;
    mobileTemplate += mobileTag;
  });

  projectData.innerHTML = desktopTemplate;
  projectsMob.innerHTML = mobileTemplate;
}

// Call the function passing the projects array to generate the projects section
generateProjectsSection(projects);

// JavaScript: Implement popup window for both mobile and desktop

const popupContainer = document.getElementById('popupContainer');

function createPopup(project) {
  const popupHTML = `
    <div class="popup">
      <div class="popup-header">
        <h2>${project.title}</h2>
        <span class="close-popup">&times;</span>
      </div>
      <div class="popup-body">
        <img class="popup-img" src="${project.image}" alt="">
        <p>${project.description}</p>
        <ul>${project.technologies.map((tech) => `<li>${tech}</li>`).join('')}</ul>
        <div class="popup-button">
          <button class="popup-btn"><a href="${project.liveLink}" target="_blank">See Live</a></button>
          <button class="popup-btn"><a href="${project.sourceLink}" target="_blank">See Source</a></button>
        </div>
      </div>
    </div>
  `;

  return popupHTML;
}

projectData.addEventListener('click', (event) => {
  if (event.target.classList.contains('view-project')) {
    const index = Array.from(projectData.querySelectorAll('.view-project')).indexOf(event.target);
    const project = projects[index];
    const popup = createPopup(project);
    popupContainer.innerHTML = popup;
    popupContainer.classList.add('show-popup');
  }
});

popupContainer.addEventListener('click', (event) => {
  if (event.target.classList.contains('close-popup')) {
    popupContainer.innerHTML = '';
    popupContainer.classList.remove('show-popup');
  }
});
```

This code creates a function called `generateProjectsSection` that dynamically generates the projects section based on the projects array. It also implements the popup window functionality for both mobile and desktop views. When the user clicks the "See Project" button, the relevant project details appear in a popup window, and when the user clicks the close (X) button, the popup disappears.


The above is a rough implementation of what is expected of you.